src/ML-RUNNER/pair_runner.cpp

```diff
@@ -0,0 +1,1381 @@
+  // In case of 2G HDNNP, we could call virial_fdotr()
+  // for performace reasons.
+  // flag must then be overwritten in init_style().
+  no_virial_fdotr_compute = 1;
+  map = nullptr;
+
+  // Additional per-atom arrays for communication
+  nmax = 0;
+  atomic_charge = nullptr;
+  hirshfeld_volume = nullptr;
+  electronegativity = nullptr;
+  lagrange_charges = nullptr;
+  screening_de_dq = nullptr;
+  de_dq = nullptr;
+  committee_storage = nullptr;
+
+  // Set commsize needed by this pairstyle.
+  comm_forward = 1;    // Forward communication (1 double per atom)
+  comm_reverse = 1;    // Reverse communication (1 double per atom)
+  commstyle = 0;
+
+  // Sum of extrapolation is zero upon initialization of this pair style
+  local_extrap_sum = 0;
+
+  // variable for output using pair compute
+  nextra = 0;
+  pvector = nullptr;
+}
+
+PairRuNNer::~PairRuNNer()
+{
+  // Deallocate member variables
+  if (allocated) {
+    memory->destroy(setflag);
+    memory->destroy(cutsq);
+    memory->destroy(atomic_charge);
+    memory->destroy(hirshfeld_volume);
+    memory->destroy(electronegativity);
+    memory->destroy(lagrange_charges);
+    memory->destroy(de_dq);
+    memory->destroy(screening_de_dq);
+    memory->destroy(committee_storage);
+    delete[] map;
+    delete[] directory;
+    delete[] pvector;
+  }
+}
+
+void PairRuNNer::compute(int eflag, int vflag)
+{
+  const bool debug = false;
+
+  int inum, jnum, ii, jj, i, j;
+  int *ilist;
+  int *jlist;
+  int *numneigh, **firstneigh;
+  int *runner_num_neigh, *runner_first_neighbor, *runner_jlist, *runner_types;
+
+  // Initializes flags, which signal if energy and virial need to be tallied.
+  // In a typical MD simulation, these two properties are only required for
+  // output purposes (reporting potential energy and pressure), so their
+  // collection across all processes (=tallying) can be skipped in all timesteps
+  // where no reporting has been requested.
+  ev_init(eflag, vflag);
+
+  double **x = atom->x;
+  double **f = atom->f;
+  double *q = atom->q;
+  // Number of atoms owned by this process.
+  int nlocal = atom->nlocal;
+  // Number of ghost atoms on this process. Ghost atoms are atoms owned by another
+  // process which are inside the neighbor list cutoff for local atoms on this
+  // process. We have to calculate the force contribution from local atoms to
+  // these ghost atoms.
+  int nghost = atom->nghost;
+  int ntotal = nlocal + nghost;
+  // Total number of atoms in the simulation box. This is mpi_sum(nlocal) across
+  // all processes.
+  int natoms = static_cast<int>(atom->natoms);
+  int *type = atom->type;
+  tagint *tag = atom->tag;
+
+  // Interface variables
+  bool lperiodic;
+  double *committee_energy, *committee_force, *committee_d_energy_d_strain, *lattice;
+  double *committee_atomic_charge, *committee_hirshfeld_volume, *committee_electronegativity,
+      *committee_hardness;
+
+  lattice = new double[9];
```

Comment:
```suggestion
  double lattice[9] = {0.0};
```
If this array is allocated and deallocated in every call, it does not need to be a class member and with the small fixed size it can be allocated on the stack.

Reply:
Done